test/common/network/listen_socket_impl_test.cc

```diff
@@ -5,20 +5,27 @@
 namespace Network {
 
 TEST(ListenSocket, All) {
+  // Test the case of a socket with given port and bind_to_port set to true.
   TcpListenSocket socket1(uint32_t(15000), true);
-  listen(socket1.fd(), 0);
+  EXPECT_EQ(1, listen(socket1.fd(), 0));
   EXPECT_EQ(15000U, socket1.localAddress()->ip()->port());
 
+  // Test the case of a socket with given tcp address and bind_to_port set to true.
   TcpListenSocket socket2("tcp://127.0.0.1:15002", true);
-  listen(socket2.fd(), 0);
+  EXPECT_EQ(1, listen(socket2.fd(), 0));
```

Comment:
According to man (2) listen: "On success, zero is returned.  On error, -1 is returned, and errno is set appropriately."